presto-main/src/main/java/com/facebook/presto/sql/planner/EqualityInference.java

```diff
@@ -61,7 +61,7 @@
         // 3) Sort the expressions alphabetically - creates a stable consistent ordering (extremely useful for unit testing)
         // TODO: be more precise in determining the cost of an expression
         return ComparisonChain.start()
-                .compare(SymbolsExtractor.extractAll(expression1).size(), SymbolsExtractor.extractAll(expression2).size())
+                .compare(VariablesExtractor.extractAllSymbols(expression1).size(), VariablesExtractor.extractAllSymbols(expression2).size())
```

Comment:
nit: Ideally we can remove all `extractAllSymbols`.

Reply:
Ya, this one better waits until we change `Expression` to `RowExpression` otherwise we are adding a lot of code to figure out the type only to throw it away. Not really worth it IMO.